dom-lub-switchToJs

dom-lud opened the pull request.

s26964-pj committed:
change to spring Security instead of Webflux
easily and more friendly

s26964-pj committed:
sprzątanie

zmiana na JS dla login/register/homepage

s26964-pj committed:
admin zmiana na JS

s26964-pj committed:
dodanie JS dla admina + uporządkowanie kodu
- usunięcie testów

s26964-pj committed:
refactor UserService to simplify user management and password encoding

s26964-pj committed:
feat: implement parent panel with schedule, grades, attendance, and remarks views

s26964-pj committed:
feat: update routing for teacher and student roles, add student API controller

Commit:
clean up

s26964-pj committed:
Remove legacy view controllers and admin templates

Deleted old view controller classes for admin, home page, parent, student, and teacher roles, as well as their associated Thymeleaf templates. Also moved SessionService to a new package, updated .gitignore to exclude node_modules, and made minor UI adjustments in static files. This refactor cleans up unused code and prepares for a new frontend or controller structure.

s26964-pj committed:
Refactor admin API and frontend for improved DTO handling

Replaces custom record DTOs in AdminApiController with generic Map-based responses for subjects, teachers, students, remarks, and grades. Updates frontend admin.js to support new DTO formats and adds filtering to grades and attendance views. Adds user details endpoint, school class update method, and improves grade saving logic. Cleans up comments and streamlines code for maintainability.

s26964-pj committed:
refactor: clean up code formatting and remove unused imports across multiple classes

s26964-pj committed:
refactor: improve code readability by adding braces for conditional statements

s26964-pj committed:
refactor: standardize code formatting by adding braces for conditional statements

s26964-pj committed:
remove extra separate

BiveGpe commented on pom.xml:
```diff
@@ -19,7 +19,7 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-webflux</artifactId>
+            <artifactId>spring-boot-starter-web</artifactId>
```
Z czego wynika ta zmiana?

dom-lud replied:
webflux był cięższy w konfiguracji i trochę zbyt mocny/zaawansowany na nasze potrzeby. Sam krzysiu ostatnio mówił żeby usunąć te jego security bo jest do dupy

BiveGpe commented on src/main/java/com/byt/freeEdu/security/Security.java:
```diff
@@ -0,0 +1,50 @@
+  /** Konfiguracja łańcucha filtrów (MVC/Servlet). */
+  public SecurityFilterChain enforcingAuthenticationWithoutCSRF(HttpSecurity http)
+      throws Exception {
+    http.csrf(csrf -> csrf.disable())
```
Disable?

dom-lud replied:
Aktualnie tak, możemy spróbować to włączyć, ale z tego co kojarzę powoduje to problemy -> na BYT też szliśmy z disable

BiveGpe commented on src/main/resources/templates/teacher/teacher_attendanceSelect.html:
```diff
@@ -1,97 +0,0 @@
-<!DOCTYPE html>
```
Testy wszystkie zostały wyjebane?

dom-lud replied:
dobrze że zauważyłeś, już robię reverta bo gdzieś musiałem je kliknąć przy commicie przez przypadek

s26964-pj committed:
Add unit tests for service layer classes

Added comprehensive JUnit and Mockito-based unit tests for AttendanceService, GradeService, RemarkService, ScheduleService, SchoolClassService, and user-related services (AdminService, ParentService, StudentService, TeacherService, UserService). These tests cover CRUD operations, exception handling, and edge cases to improve code reliability and maintainability.

dom-lud committed:
Migrate controllers to reactive WebFlux

Refactored all REST controllers to use Project Reactor types (Mono, Flux) and reactive Spring WebFlux annotations. Updated pom.xml to use spring-boot-starter-webflux instead of spring-boot-starter-web. This enables non-blocking, reactive endpoints throughout the API, replacing ResponseEntity and synchronous return types with reactive equivalents.

dom-lud committed:
rollback

dom-lud committed:
Merge pull request #40 from Free-edu-org/text-webflux

s26964-pj committed:
Reformat controllers and security classes for consistency

Unified code style across all controller and security classes by removing extra whitespace, aligning indentation, and standardizing constructor and method formatting. No functional changes were made; this improves readability and maintainability.

s26964-pj committed:
refactor: reorganize imports and enhance code readability across controllers and models

s26964-pj committed:
refactor: standardize exception handling by adding braces for conditional statements

dom-lud committed:
Merge branch 'master' into dom-lub-switchToJs
